Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts

```diff
@@ -0,0 +1,68 @@
+import { AzureRmWebAppDeploymentProvider } from './AzureRmWebAppDeploymentProvider';
+import tl = require('vsts-task-lib/task');
+import { FileTransformsUtility } from '../operations/FileTransformsUtility';
+import { DeployWar } from '../operations/WarDeploymentUtilities';
+import * as Constant from '../operations/Constants';
+import { WebDeployUtility } from '../operations/WebDeployUtility';
+import { AzureAppServiceUtility } from '../operations/AzureAppServiceUtility';
+import { Package } from 'webdeployment-common/packageUtility';
+import * as ParameterParser from '../operations/parameterparser'
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var deployUtility = require('webdeployment-common/utility.js');
+var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
+const runFromZipAppSetting: string = '-WEBSITE_RUN_FROM_ZIP 1';
+
+export class WindowsWebAppZipDeployProvider extends AzureRmWebAppDeploymentProvider{
+    
+    private zipDeploymentID: string;
+    private runFromZip: boolean;
+ 
+    public async DeployWebAppStep() {
+        var webPackage = await FileTransformsUtility.applyTransformations(this.taskParams.Package.getPath(), this.taskParams);
+
+        if(this.taskParams.DeploymentType === "zipDeploy") {
+
+            var _isMSBuildPackage = await this.taskParams.Package.isMSBuildPackage(); 
+            if(_isMSBuildPackage) {
+                throw Error(tl.loc("Publishusingzipdeploynotsupportedformsbuildpackage"));
+            }
+            else if(this.taskParams.VirtualApplication) {
+                throw Error(tl.loc("Publishusingzipdeploynotsupportedforvirtualapplication"));
+            }
+
+            await this.deployUsingZipDeploy(webPackage, this.taskParams.UseRunFromZip);
+        }
+        else if(this.taskParams.ScriptType) {
+            await this.deployUsingZipDeploy(webPackage, false);
+        }
+        else {
+            await this.deployUsingZipDeploy(webPackage, true);
+        }
+
+        await this.PostDeploymentStep();
```

Comment:
no post deployment script in RunFromZip

Reply:
There are two cases when runFromZip will be used:
- When user explicitly selects from the task, in that case we dont allow user to add post deployment script.
- When we do auto-detect, as you see the logic if the user adds the deployment script we are calling zipdeploy with runFromZip flag value as "false"
Do you see any other scenario that I am missing?

Reply:
we might want to fallback  to puer zipdeploy on auto-detect when Post deployment script is present. @kmkumaran , @azooinmyluggage  , can you please share your thoughts?